### PR TITLE
add item screen fix

### PR DESCRIPTION
**Pull request by EntyR**

### CODE REVIEW

**Comment by AlexeyKolosov** [comment]
Please, describe the case, what this bug was. (maybe add this to a new bug issue)
1. Steps to reproduce
2. Expected behavior
3. Actual behavior

File: app/src/main/java/com/harman/roomdbapp/app/ui/viewmodel/AddNumberViewModel.kt [hunk 1]
```diff
@@ -19,10 +19,11 @@ class AddNumberViewModel(
     private val _textState = MutableLiveData<AddNumberState>(AddNumberState.AddRandom)
     val textState: LiveData<AddNumberState> = _textState
 
-    fun switchState() {
-        if (_textState.value == AddNumberState.AddCustom)
+    fun switchState(text: String) {
+        if (text.isEmpty() && textState.value == AddNumberState.AddCustom)
```
Comment (AlexeyKolosov) [inline app/src/main/java/com/harman/roomdbapp/app/ui/viewmodel/AddNumberViewModel.kt]: please try to modify logics based only on text, not on the previous state, then add tests

File: app/src/main/java/com/harman/roomdbapp/app/ui/viewmodel/AddNumberViewModel.kt [hunk 2]
```diff
@@ -20,9 +20,9 @@ class AddNumberViewModel(
     val textState: LiveData<AddNumberState> = _textState
 
     fun switchState(text: String) {
-        if (text.isEmpty() && textState.value == AddNumberState.AddCustom)
+        if (text.isEmpty())
             _textState.value = AddNumberState.AddRandom
-        else if(textState.value == AddNumberState.AddRandom)
+        else if(text.isNotEmpty())
```
Comment (AlexeyKolosov) [inline app/src/main/java/com/harman/roomdbapp/app/ui/viewmodel/AddNumberViewModel.kt]: I `else` is enough
```
 _textState.value = if (text.isEmpty()) AddNumberState.AddRandom else AddNumberState.AddCustom

File: app/src/main/java/com/harman/roomdbapp/app/ui/viewmodel/AddNumberViewModel.kt (hunk 1 above)
Comment (EntyR) [reply app/src/main/java/com/harman/roomdbapp/app/ui/viewmodel/AddNumberViewModel.kt]: done

**Review by AlexeyKolosov — APPROVED** [review]

**Pull request merged**